Makefile

```diff
@@ -75,33 +75,24 @@ INSTALL_DIR = $(INSTALL) -d
 INSTALL_PROGRAM = $(INSTALL) -m755
 INSTALL_DATA = $(INSTALL) -m644
 
-# Toolchain
-MSBUILD = msbuild -verbosity:m -nologo
-
 # Enable 32 bit builds while generating the windows installer
 WIN32 = false
 
+TARGETPLATFORM = linux-x64
+
 # program targets
 VERSION     = $(shell git name-rev --name-only --tags --no-undefined HEAD 2>/dev/null || echo git-`git rev-parse --short HEAD`)
 
-# dependencies
-UNAME_S := $(shell uname -s)
-ifeq ($(UNAME_S),Darwin)
-os-dependencies = osx-dependencies
-else
-os-dependencies = linux-dependencies
-endif
-
 check-scripts:
 	@echo
 	@echo "Checking for Lua syntax errors..."
 	@luac -p $(shell find mods/*/maps/* -iname '*.lua')
 	@luac -p $(shell find lua/* -iname '*.lua')
 
-check: dependencies
+check:
 	@echo
 	@echo "Compiling in debug mode..."
-	@$(MSBUILD) -t:build -p:Configuration=Debug
+	@dotnet build --configuration Debug
```

Comment:
This makes `dotnet` a dependency for compiling on all platforms - we'll need to either update INSTALL.md to reflect this or, (IMO better to reduce scope here) keep this as $(MSBUILD) until the PR that adds net core support.